scripts/ci/publish.ts

```diff
@@ -437,43 +420,38 @@ function getMaxPatchVersionIncrement(versions: string[]): number {
   return Math.max(...increments, 0)
 }
 
-// TODO: we can simplify this into one line (cc Joël)
-async function getAllVersions(packages: Packages, channel: string, prefix: string): Promise<string[]> {
-  return unique(
-    flatten(
-      await pMap(
-        Object.values(packages).filter((p) => p.name !== '@prisma/integration-tests'),
-        async (pkg) => {
-          if (pkg.name === '@prisma/integration-tests') {
-            return []
-          }
-          const pkgVersions = [] as string[]
-          if (pkg.version.startsWith(prefix)) {
-            pkgVersions.push(pkg.version)
-          }
-          const remoteVersionsString = await runResult('.', `npm info ${pkg.name} versions --json`)
-
-          const remoteVersions = JSON.parse(remoteVersionsString)
-
-          for (const remoteVersion of remoteVersions) {
-            if (
-              remoteVersion.includes(channel) &&
-              remoteVersion.startsWith(prefix) &&
-              !pkgVersions.includes(remoteVersion)
-            ) {
-              pkgVersions.push(remoteVersion)
-            }
-          }
-          return pkgVersions
-        },
-        { concurrency: 3 }, // Let's not spam npm too much
-      ),
-    ),
-  )
+export async function getLatestVersionPublishedFor(pkgs: Packages, channel: string, prefix: string): Promise<string[]> {
+  // We check the versions for the `@prisma/debug` package
+  // Why?
+  // Because `@prisma/debug` is the first package that will be published
+  // So if npm fails to publish one of the packages,
+  // we cannot republish on the same version on a next run
+  const pkg = pkgs['@prisma/debug']
```

Comment:
So if we do eg a Prisma `4.10` release, if it fails in the middle for whatever reason, we re-trigger it and it will do `4.11` next iirc?

Reply:
No this is done in `getNextMinorStable`, I think I want to rename this and add a comment